### PR TITLE
feat(dashboard-templates): Hook up the Add button

### PR DESCRIPTION
**Pull request by wmak**
- This adds functionality to the add dashboard button
   - This uses the duplicate param on create to avoid re-using the same
     dashboard name

### CODE REVIEW

File: static/app/views/dashboardsV2/manage/index.tsx [hunk 1]
```diff
@@ -220,6 +222,13 @@ class ManageDashboards extends AsyncView<Props, State> {
     });
   }
 
+  onAdd(dashboard: DashboardDetails) {
+    const {organization, api} = this.props;
```
Comment (wmak) [inline static/app/views/dashboardsV2/manage/index.tsx:226]: TODO: forgot to add analytics here

**Review by shruthilayaj — APPROVED** [review]

File: static/app/views/dashboardsV2/manage/index.tsx [hunk 2]
```diff
@@ -220,6 +222,18 @@ class ManageDashboards extends AsyncView<Props, State> {
...
+    trackAdvancedAnalyticsEvent('dashboards_manage.templates.add', {
+      organization,
+      dashboard_id: dashboard.id,
+    });
+
+    createDashboard(api, organization.slug, dashboard, true).then(() =>
+      this.onDashboardsChange()
+    );
```
Comment (evanpurkhiser) [inline static/app/views/dashboardsV2/manage/index.tsx:234]: why not make this function async, so we can


```suggestion
    await createDashboard(api, organization.slug, dashboard, true)
    this.onDashboardsChange()
```

File: static/app/views/dashboardsV2/manage/templateCard.tsx [hunk 3]
```diff
@@ -28,7 +29,11 @@ function TemplateCard({title, widgetCount, onPreview}: Props) {
         <StyledButton priority="default" onClick={onPreview}>
           {t('Preview')}
         </StyledButton>
-        <StyledButton priority="primary" icon={<IconAdd size="xs" isCircled />}>
+        <StyledButton
+          priority="primary"
+          onClick={onAdd}
+          icon={<IconAdd size="xs" isCircled />}
+        >
```
Comment (evanpurkhiser) [inline static/app/views/dashboardsV2/manage/templateCard.tsx:36]: let's add a `label`?

Comment (evanpurkhiser) [reply static/app/views/dashboardsV2/manage/templateCard.tsx:36]: Guess it didn't already have one 🙃

Comment (wmak) [reply static/app/views/dashboardsV2/manage/templateCard.tsx:37]: Added a label :+1:

**Pull request merged**